lib/rubygems/version.rb

```diff
@@ -384,6 +384,24 @@ def <=>(other)
       return lhs <=> rhs
     end
 
+    lhs = lhsegments[i]
```

Comment:
Is it worth a short comment in here to describe what's the benefit in here?

Reply:
I'm usually a proponent of using `git blame` to answer "why this code" but I recognize others prefer comments 😅. I'd be happy to add one if you think it would be beneficial.

Reply:
git blame sounds ok also